core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -1548,4 +1565,44 @@ public static boolean isFaultTolerantExecutionEventDriverSchedulerEnabled(Sessio
     {
         return session.getSystemProperty(FAULT_TOLERANT_EXECUTION_EVENT_DRIVEN_SCHEDULER_ENABLED, Boolean.class);
     }
+
+    private static <T> PropertyMetadata<T> addBounds(PropertyMetadata<T> property, boolean hasDefaultLowerBound, boolean hasDefaultUpperBound)
+    {
+        if (!hasDefaultLowerBound && !hasDefaultUpperBound) {
+            return property;
+        }
+        if (!COMPARABLE_JAVA_TYPES.contains(property.getJavaType())) {
+            throw new TrinoException(INVALID_SESSION_PROPERTY, format("Session property %s of type %s does not support lower/upper bound", property.getName(), property.getSqlType()));
+        }
+        return new PropertyMetadata<>(
+                property.getName(),
+                property.getDescription(),
+                property.getSqlType(),
+                property.getJavaType(),
+                property.getDefaultValue(),
+                property.isHidden(),
+                object -> {
+                    T value = property.decode(object);
+                    @SuppressWarnings("unchecked")
+                    Comparable<T> comparable = (Comparable<T>) value;
```

Comment:
That would need to to use Trino type comparison operator.

Reply:
Durations are encoded as `VARCHAR` so don't think that'll work? Anyway irrelevant now that I simplified this.